### PR TITLE
Neutral geonames

### LINKED ISSUE #1808: A serious question about Ukraine`s sovereignty
Dear Developers,

I am a user from Ukraine. I would like to inform you that starting with version v0.19.2 (29 Sep 2019), the cities of Crimea are signed as the Russian Federation. What happened on June 22, 2019 (that's the release date of the previous version)? Another "referendum" in Crimea? It is especially surprising that there is no information about this change in the Changelog: https://github.com/Stellarium/stellarium/releases/tag/v0.19.2. Your decision defies all logic.

Under international law, the Crimean peninsula is part of Ukraine. United Nations General Assembly Resolution 68/262: https://en.wikipedia.org/wiki/United_Nations_General_Assembly_Resolution_68/262

Best Regards,
Mykhaylo

### PR DESCRIPTION
**Pull request by alex-w**
### Description
This pull request contains implementation of using politically neutral geographic locations. I use United Nations geoscheme (UN M49) as definitions for geographic regions. Code for backward conpatibility for countries also added (to use in landscapes and sceneries).

Fixes #1808, #1135, #1125, #894

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by gzotti** [comment]
In terms of recent quibbles, this is a good approach, thank you! Astronomy programs need location names just for identification, not for political statements. I must have a deeper look at it!

**Comment by gzotti** [comment]
Not sure what the previous state was: when I am on Tethys, opening the region list shows the regions and placenames on earth. 
Maybe the location panel still needs that really thorough rewrite, with extraterrestrial locations taken from the planet features (NomenclatureItems) list. But if possible, the region list should be adapted when observer moves to another planet.

**Comment by alex-w** [comment]
> Not sure what the previous state was: when I am on Tethys, opening the region list shows the regions and placenames on earth.
> Maybe the location panel still needs that really thorough rewrite, with extraterrestrial locations taken from the planet features (NomenclatureItems) list. But if possible, the region list should be adapted when observer moves to another planet.

The behavior of region selector is same as was in country selector. The regions for extraterrestrial locations may be added for artificial objects only IMHO.

**Comment by gzotti** [comment]
OK. Else, it works well, or I found no bug so far. But this means, the location sections will have to include a "region" code now, with only few possible entries. The full list should go into the Guide, e.g. into ch.7, and fig. 4.2 replaced. We can add an extended comment or footnote about this change to section 4.2.

**Comment by alex-w** [comment]
> But this means, the location sections will have to include a "region" code now, with only few possible entries. 

I've just replaced the country code by the "region code" (the region code has 2 digits to backward compatibility in one hand and to the strict segregation in other hand).

> The full list should go into the Guide, e.g. into ch.7, and fig. 4.2 replaced. We can add an extended comment or footnote about this change to section 4.2.

Why? The landscapes (`county` item in ini data) and 3D sceneries (`county` item in ini data) hasn't changes and no need updating Chapter 7. Replacement the figure 4.2 and footnote for section 4.2 are needed of course.

**Comment by alex-w** [comment]
> Not sure what the previous state was: when I am on Tethys, opening the region list shows the regions and placenames on earth.

Done. The list of regions depends by planet now.

P.S. I’ll update regions for extraterrestrial locations tomorrow. Maybe we should update our list of extraterrestrial locations also?

**Review by gzotti — CHANGES_REQUESTED** [review]
in file base_locations.txt, line 25086
Luna 21	USSR	Mare Sedenitatis	L	0	25.51N	30.38E	134	0		Moon	moon

should be "Mare serenitatis"

**Comment by gzotti** [comment]
When I am on the moon and press "Reset locations list", the list is filled with all known locations. This may have been already like this, but it may be more useful to filter by current planet.

**Comment by alex-w** [comment]
> When I am on the moon and press "Reset locations list", the list is filled with all known locations. This may have been already like this, but it may be more useful to filter by current planet.

No, this is correct behavior - this button reset all filters

**Comment by alex-w** [comment]
I think the PR is ready to merge

**Comment by gzotti** [comment]
I have just seen the regions-geoscheme.tab. Probably we should keep that expandable by having codes <100 on Earth, 1xx=Mercury, 2xx=Venus, 3xx=Moon, 4xx=Mars etc. For outer planets' moons, a scheme like 51xx for Io, 61xx for Mimas etc. would be consistent. This would allow generous 100 regions per object which should be OK for rough guidance (probably only 2 large regions have been named on Iapetus.. :-). 

The current list is else incomplete and also inconsistent: Mare Moscoviense, M. Orientale, M. Smythii etc are missing, and a few craters are given. Who knows crater Mandel'shtam? Why not include Tsiolkowski or Plato? Is Xanadu the only named region on Titan? etc.etc. Of course, this overlaps also with the planet features (NomenclatureItem), which may also be extended to give hierarchical regions. ("Crater XY in Apennines highland")

**Comment by alex-w** [comment]
> I have just seen the regions-geoscheme.tab. Probably we should keep that expandable by having codes <100 on Earth, 1xx=Mercury, 2xx=Venus, 3xx=Moon, 4xx=Mars etc. For outer planets' moons, a scheme like 51xx for Io, 61xx for Mimas etc. would be consistent. This would allow generous 100 regions per object which should be OK for rough guidance (probably only 2 large regions have been named on Iapetus.. :-).

Maybe. This is acceptable, but what you want to get in final?

> The current list is else incomplete and also inconsistent: Mare Moscoviense, M. Orientale, M. Smythii etc are missing, and a few craters are given. Who knows crater Mandel'shtam? Why not include Tsiolkowski or Plato? Is Xanadu the only named region on Titan? etc.etc. Of course, this overlaps also with the planet features (NomenclatureItem), which may also be extended to give hierarchical regions. ("Crater XY in Apennines highland")

Do you want to include all tens of thousand records for Earth locations into Nomenclature items and all consolidated list export to regions? But for what? Mare Orientale is not in the list of reagions, because this mare contains no artificial objects. Let's imagine, than we include Tsiolkowski into list of regions and what this fact will give us? I can answer - nothing. Adding Tsiolkowski into list of locations of course can give us some meaning, but in this case we should expand list by over 175000 items and all of those 200000+ items should be available in two places (Nomenclature and Locations) and translatable.

**Comment by gzotti** [comment]
I mainly noted the current limitation which grew out of the current list. Several contestants are sending further landers to Mars&Moon, maybe to other Moons of the outer planets, so the list must be able to grow, and the scheme I proposed should allow a better structure from the beginning.  I don't know if some Lunar atlas has already devised a "region" partition of the Moon, but having just a few little craters as "regions" also looks wrong to me. So, a split into Maria and Terrae/highlands seems natural for the Moon, and then the spacecraft landing sites may be given more appropriate "regions". 

The existing Nomenclature list is of course also pretty interesting and inviting to load into the location panel when on the Moon or Mars. But this is another task.

**Comment by alex-w** [comment]
Mars and Venus already have separation on giant regions and I can use it. Now I need check my atlases on the Moon. :)

**Comment by alex-w** [comment]
OK, I think the changes is complete and PR is ready to merge

File: data/regions-geoscheme.tab [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+311	Moon	Sea of Islands
+312	Moon	Sea of the Edge
+313	Moon	Sea of Moscow
+314	Moon	Sea of Nectar
+315	Moon	Sea of Clouds
+316	Moon	Eastern sea
+317	Moon	Sea of Serenity
+318	Moon	Smyth's Sea
+319	Moon	Foaming Sea
+320	Moon	Sea of Tranquility
```
Comment (gzotti) [inline data/regions-geoscheme.tab:161]: better: Tranquillity

**Comment by gzotti** [comment]
OK, looks much better. For consistency with other extraterrestrial names, we may still consider using the traditional Latin names also on the Moon, at least for the Maria. @sushoff ?

**Comment by alex-w** [comment]
> OK, looks much better. For consistency with other extraterrestrial names, we may still consider using the traditional Latin names also on the Moon, at least for the Maria. @sushoff ?

I prefer to use English names for regions of the Moon to avoid mixed 2 languages. Or we should translate all regions of the Moon to latin.

File: data/regions-geoscheme.tab (hunk 1 above)
Comment (alex-w) [reply data/regions-geoscheme.tab]: Ok

**Comment by alex-w** [comment]
Is it ok?

**Comment by gzotti** [comment]
For me it's OK. @sushoff what's your opinion here?

**Comment by sushoff** [comment]
> > OK, looks much better. For consistency with other extraterrestrial names, we may still consider using the traditional Latin names also on the Moon, at least for the Maria. @sushoff ?
> 
> I prefer to use English names for regions of the Moon to avoid mixed 2 languages. Or we should translate all regions of the Moon to latin.

yes, I think the lunar maria are comonly known with both, Latin name and its translation (at least among moon observers) - therefroe, I think, in this particular Latin addition would be fine while in all other cases the English-only solution should work well

**Comment by axd1967** [comment]
another example where vendor branches can simplify code management.

**Pull request merged**